### ISSUE TITLE
Ruby 3 keyword args compatibility?

### ISSUE THREAD

**Issue by cben** [abonas/kubeclient#431]
https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/
Apparently it's possible to write forward-compatible code now.  Can test on 2.7 to get warnings.
Low priority, but testing and PRs always welcome.

**Comment by cben** [comment]
2.7 starts showing some warnings about this, which we can start fixing, e.g.
```
lib/kubeclient.rb:27: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
lib/kubeclient/common.rb:61: warning: The called method `initialize_client' is defined here

lib/kubeclient/common.rb:232: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
lib/kubeclient/common.rb:355: warning: The called method `delete_entity' is defined here
``` 
(see #432 about other issues in 2.7)

**Comment by cben** [comment]
Hopefully this got fixed by #432.  (at least the warnings are gone)
